### PR TITLE
Support Parsing `DROP VIEW` in Oracle

### PR DESCRIPTION
**Pull request by everly-gif**
Refers #6480.

Changes proposed in this pull request:

- Supports parsing `DROP VIEW` in Oracle
- Adds Tests
- Corrects files

✅ Builds Locally

### CODE REVIEW

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 [hunk 1]
```diff
@@ -43,6 +43,10 @@ dropIndex
     : DROP INDEX indexName ONLINE? FORCE? ((DEFERRED|IMMEDIATE) INVALIDATION)?
     ;
 
+dropView
+    : DROP VIEW (schemaName DOT_)? viewName (CASCADE CONSTRAINTS)?
```
Comment (Liangda-w) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:47]: `viewName` is defined in `BaseRule.g4` as the following
```
viewName
    : (owner DOT_)? name
    ;
```
so you can leave out the `(schemaName DOT_)?` here

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/oracle/ddl/OracleDropViewStatement.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import lombok.ToString;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.AbstractSQLStatement;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.ddl.DDLStatement;
+import org.apache.shardingsphere.sql.parser.sql.dialect.statement.oracle.OracleStatement;
+
+/**
+ * Oracle drop view statement.
+ */
+@ToString
+public final class OracleDropViewStatement extends AbstractSQLStatement implements DDLStatement, OracleStatement {
```
Comment (Liangda-w) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/oracle/ddl/OracleDropViewStatement.java:29]: `DropViewStatement.java` exists, so you can extend it directly

**Review by Liangda-w — CHANGES_REQUESTED** [review]

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/oracle/ddl/OracleDropViewStatement.java (hunk 2 above)
Comment (everly-gif) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/oracle/ddl/OracleDropViewStatement.java]: Okay, have made the changes

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 (hunk 1 above)
Comment (everly-gif) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4]: Oh yes 👍, Made the changes.

**Review by Liangda-w — APPROVED** [review]
LGTM, thank you 👍

**Comment by strongduanmu** [comment]
Hi @everly-gif, thank you for your great contribution. Can you submit a new pr to support create view and alter view statements?

**Comment by everly-gif** [comment]
@strongduanmu Sure ! You should soon see two PRs from me :)

**Comment by strongduanmu** [comment]
> @strongduanmu Sure ! You should soon see two PRs from me :) Also do you want me to do it for Oracle or PostgreSQL? I'm assuming Oracle.

@everly-gif Yes, it's Oracle database. You can try to support Oracle view statements parse first, and then refer pr #16771 to adapt Oracle views so that they can execute normally.

**Comment by everly-gif** [comment]
The `create view` [doc](https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/CREATE-VIEW.html#GUID-61D2D2B4-DACC-4C7C-89EB-7E50D9594D30) grammar is very complex. However I have tried to define the grammar and the antlr4 plugin **throws no error**. But the `shardingsphere-sql-parser-dialect` module has test failures. The only error I receive is` there is an error in your SQL syntax`.  So I haven't been able to debug much. 

Here's the grammar I wrote,

In DDLStatement.g4
```
createView
    : CREATE (OR REPLACE)? ((NO)? FORCE)? ( EDITIONING | EDITIONABLE (EDITIONING)? | NONEDITIONABLE)? VIEW viewName
    createSharingClause ( LP_ (aliasClause (COMMA_ aliasClause)* | outOfLineConstraint) RP_  | createObjectViewClause | 
    createXMLTypeViewClause)? (DEFAULT COLLATION collationName)? (BEQUEATH (CURRENT_USER | DEFINER))? AS subquery 
    (subqueryRestrictionClause)? (CONTAINER_MAP | CONTAINERS_DEFAULT)?
    ;

createObjectViewClause
    : OF objectName (
    | WITH OBJECT (IDENTIFIER | ID) (
    | DEFAULT
    | attributeNames )
    | UNDER superViewName )
    ( LP_ ( outOfLineConstraint (COMMA_ outOfLineConstraint)* | attributeName inlineConstraint* ) RP_ )?
    ;

createXMLTypeViewClause
    : OF XMLTYPE xmlSchemaSpecClause? WITH OBJECT (IDENTIFIER | ID) (DEFAULT | expressionList)
    ;   

aliasClause
    : (VISIBILE | INVISIBLE)? (inlineConstraint*)?
    ;

subqueryRestrictionClause
    : WITH (
    | READ ONLY
    | CHECK OPTION
    ) (CONSTRAINT constraintName)?
    ;
```
In OracleKeyword.g4

```
EDITIONING
    : E D I T I O N I N G
    ;

ID
    : I D
    ;  

BEQUEATH
    : B E Q U E A T H
    ;

```
In Baserule.g4

```
superViewName
    : (owner DOT_)? name
    ;

attributeNames
    : LP_ attributeName (COMMA_ attributeName)* RP_
    ;
```


Can you please guide me with what is the issue here.

**Comment by everly-gif** [comment]
The `create view` [doc](https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/CREATE-VIEW.html#GUID-61D2D2B4-DACC-4C7C-89EB-7E50D9594D30) grammar is very complex. However I have tried to define the grammar and the antlr4 plugin **throws no error**. But the `shardingsphere-sql-parser-dialect` module has test failures. The only error I receive is` there is an error in your SQL syntax`.  So I haven't been able to debug much. 

Here's the grammar I wrote,

In DDLStatement.g4
```
createView
    : CREATE (OR REPLACE)? ((NO)? FORCE)? ( EDITIONING | EDITIONABLE (EDITIONING)? | NONEDITIONABLE)? VIEW viewName
    createSharingClause ( LP_ (aliasClause (COMMA_ aliasClause)* | outOfLineConstraint) RP_  | createObjectViewClause | 
    createXMLTypeViewClause)? (DEFAULT COLLATION collationName)? (BEQUEATH (CURRENT_USER | DEFINER))? AS subquery 
    (subqueryRestrictionClause)? (CONTAINER_MAP | CONTAINERS_DEFAULT)?
    ;

createObjectViewClause
    : OF objectName (
    | WITH OBJECT (IDENTIFIER | ID) (
    | DEFAULT
    | attributeNames )
    | UNDER superViewName )
    ( LP_ ( outOfLineConstraint (COMMA_ outOfLineConstraint)* | attributeName inlineConstraint* ) RP_ )?
    ;

createXMLTypeViewClause
    : OF XMLTYPE xmlSchemaSpecClause? WITH OBJECT (IDENTIFIER | ID) (DEFAULT | expressionList)
    ;   

aliasClause
    : (VISIBILE | INVISIBLE)? (inlineConstraint*)?
    ;

subqueryRestrictionClause
    : WITH (
    | READ ONLY
    | CHECK OPTION
    ) (CONSTRAINT constraintName)?
    ;
```
In OracleKeyword.g4

```
EDITIONING
    : E D I T I O N I N G
    ;

ID
    : I D
    ;  

BEQUEATH
    : B E Q U E A T H
    ;

```
In Baserule.g4

```
superViewName
    : (owner DOT_)? name
    ;

attributeNames
    : LP_ attributeName (COMMA_ attributeName)* RP_
    ;
```


Can you please guide me with what is the issue here.

**Pull request merged**